### PR TITLE
Support Java toolchains in Gradle plugin

### PR DESCRIPTION
**Pull request by vlsi**
#### Summary

It enables selecting Java launcher for executing sigstore-java client.

#### Release Note

NONE

#### Documentation

NONE

### CODE REVIEW

**Comment by vlsi** [comment]
This will conflict with https://github.com/sigstore/sigstore-java/pull/301, so it will go after.

File: sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/tasks/SigstoreSignFilesTask.kt [hunk 1]
```diff
@@ -159,10 +175,28 @@ abstract class SigstoreSignFilesTask : DefaultTask() {
     protected fun sign() {
         // Ensure task holds a lock, so no other signign tasks executes concurrently
```
Comment (loosebazooka) [inline sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/tasks/SigstoreSignFilesTask.kt:176]: I think this comment maybe is unncessary now.

**Review by loosebazooka — APPROVED** [review]

File: sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/tasks/SigstoreSignFilesTask.kt (hunk 1 above)
Comment (vlsi) [reply sigstore-gradle/sigstore-gradle-sign-base-plugin/src/main/kotlin/dev/sigstore/sign/tasks/SigstoreSignFilesTask.kt:176]: It is a comment to signingservice.get
Serialization is needed so Gradle does not spawn multiple worker processes as each process would perform its own OIDC dance

**Comment by loosebazooka** [comment]
do you want to merge this?

**Comment by vlsi** [comment]
>Would the gradle sigstore plugin fail to load before we hit this? since it's built with java11? oh jk, it's pure kotlin?

The toolchain support would enable users to select different JDK.
For instance, if someone wants to launch Gradle builds with Java 8, they will need a way to configure sigstore task to run with Java 11 or 17.

**Pull request merged**